Add the concept of streams to Code Intelligence (#4569)
sourcegraph/handbook: content/departments/engineering/teams/code-intelligence/index.md

- In the near term we want Code intelligence to provide seamless, out-of-the-box, precise code navigation for languages that cover 90% of the market usage.
- We want our code navigation to reach IDE feature parity, while offering the option of plugging into developers’ favorite IDEs.
- In the longer term, we envision building a global knowledge graph that accurately maps the entire code universe.
- Our complete strategy is available at [Code intelligence strategy](../../../../strategy-goals/strategy/code-intelligence/index.md)
- See [Code Intelligence history](./history.md) for our journey so far!

### Video walkthrough

<div style="position: relative; padding-bottom: 56.25%; height: 0;"><iframe src="https://www.loom.com/embed/fcaddfd333da487cb526a4fc99ead803" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen style="position: absolute; top: 0; left: 0; width: 100%; height: 100%;"></iframe></div>

## Members

{{generator:product_team.code_intelligence}}

## How we work

> On Slack, use the `#code-intel` channel, tag `@code-intel-support` for any help or use the `@codeintel` handle to notify the whole team. As detailed below in the streams section we do have several additional internal rooms however we expect almost all questions and conversations with external teams/departments to take place in `#code-intel`.

Reviewer: Are we still #code-intel even after the new marketing efforts?

Author: Correct, internally we still are.

## Streams

Code Intelligence comprises a wide cognitive domain. To be effective and feel high ownership over a wide area is difficult. To tackle this we as a team have adopted the concept of streams (also known as 'soft-teams'). A stream is a smaller set of members from within the wider team usually centered around a specific skill set or cognitive space. Each stream owns a distinct area.

The aim is for streams to have the following characteristics:

- Long lived, these are not ephemeral groupings.
- Allow deep technical ownership and collaboration.
- Building technical roadmaps with a view on the long term.
- Firm but not brittle boundaries (people can operate across many streams when needed).
- Allow us in the future to scale and create distinct teams with minimal overhead.
- People feel 100% supported to comment, help and collaborate across the whole team.

There is no "that's not my responsibility" with the use of streams, we just have certain areas that are best filled by specific skill sets.

### Language Tools

> - Slack channel: #language-tools

Reviewer 2: ```suggestion
- Slack channel: #language-tools-internal
```

> - Github board --> https://github.com/orgs/sourcegraph/projects/211/views/51

Reviewer 2: (Apologies for pedantry)

```suggestion
- GitHub board: https://github.com/orgs/sourcegraph/projects/211/views/51
```

> ##### Current members

Reviewer 3: Should we link to their handbook page or include full names?

Author: I can change to full names, I'll see if I can link to the handbook page for people but that data is generated from yaml files dynamically.

- Olaf
- TJ
- Varun

##### Focus and ownership

> The Language Tools stream focus on tooling specific to programming languages and their counterpart ecosystems to provide the best Precise Intelligence experience. The aim is to have a symbiotic relationship

Reviewer 3: ```suggestion
The Language Tools stream focus on tooling specific to programming languages and their counterpart ecosystems to provide the best precise code navigation experience. The aim is to have a symbiotic relationship
```

with the Language Platform team in that, Language Tools will provide the specific logic per problem (think syntax highlighting) whilst Language Platform will expose an agnostic/programatic way to scale this. Specific areas of ownership include:

- Precise Intelligence
  - [SCIP](https://github.com/sourcegraph/scip#readme)

>   - SCIP indexers

Reviewer 3: ```suggestion
  - [SCIP indexers](https://docs.sourcegraph.com/code_intelligence/references/indexers)
```

  - LSIF indexers
  - Community indexer contributions
- Auto-inference behavior for indexing projects
- Syntax highlighting (including the syntect service)

> - Package integrations (Maven,NPM, PyPI, Crates)

Reviewer 4: ```suggestion
- Package integrations (Maven, NPM, PyPI, Crates)
```

Reviewer 2: (Apologies for pedantry again but:) It should be `npm` not NPM.

Author: No need to apologise, will change to `npm` :)

### Language Platform

> Language Platform's focus is split into two distinct areas, the creation of data (invoking SCIP indexers via Executors _or_ generating symbols) and the productionalization of said data (exposing the data in a queryable, performant and scalable way). Language Platform aims to empower two users, providing a programatic/pluggable system for the Language Tools stream and for downstream clients to build ontop of the Code Graph data (the Code Navigation stream and also additional product teams).

Reviewer 3: ```suggestion
Language Platform's focus is split into two distinct areas, the creation of data (invoking SCIP indexers via Executors _or_ generating symbols) and the productionalization of said data (exposing the data in a queryable, performant and scalable way). Language Platform aims to empower two users, providing a programatic/pluggable system for the Language Tools stream and for downstream clients to build on top of the Code Graph data (the Code Navigation stream and also additional product teams).
```

> - Slack channel: #language-platform

Reviewer 2: ```suggestion
- Slack channel: #language-platform-and-navigation-internal
```

> - Github board --> https://github.com/orgs/sourcegraph/projects/211/views/48

Reviewer 2: ```suggestion
- GitHub board: https://github.com/orgs/sourcegraph/projects/211/views/48
```

##### Current members

- Eric
- Noah
- Chris
- Cesar

##### Focus and ownership

- Code Intelligence backend
  - Unified Intelligence API
  - Executors & Auto-indexing
  - Code Intel DB
  - Code Intelligence observability
- Symbol generation
  - Rockskip
  - Symbols service
- Code Intel extensions/code intellify

### Code Navigation

Code Navigation owns the UI/UX customer facing code navigation experience predominately focusing on the file view.

> - Slack channel: #code-navigation

Reviewer 2: ```suggestion
- Slack channel: #code-navigation-internal
```

> - Github board --> https://github.com/orgs/sourcegraph/projects/211/views/21

Reviewer 2: ```suggestion
- GitHub board: https://github.com/orgs/sourcegraph/projects/211/views/21
```
